plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -192,12 +192,35 @@ chain_plugin::chain_plugin()
    app().register_config_type<eosio::chain::db_read_mode>();
    app().register_config_type<eosio::chain::validation_mode>();
    app().register_config_type<chainbase::pinnable_mapped_file::map_mode>();
+   app().register_config_type<eosio::chain::wasm_interface::vm_type>();
 }
 
 chain_plugin::~chain_plugin(){}
 
 void chain_plugin::set_program_options(options_description& cli, options_description& cfg)
 {
+   // build wasm_runtime help text
+   std::string wasm_runtime_opt = "Override default WASM runtime (";
+   std::string wasm_runtime_desc;
+   std::string delim;
+#ifdef EOSIO_EOS_VM_JIT_RUNTIME_ENABLED
+   wasm_runtime_opt += " \"eos-vm-jit\"";
+   wasm_runtime_desc += "\"eos-vm-jit\" : A WebAssembly compiler that takes WASM and generates native code using a just in time compiler.\n";
+   delim = ", ";
+#endif
+
+#ifdef EOSIO_EOS_VM_RUNTIME_ENABLED
+   wasm_runtime_opt += delim + "\"eos-vm\"";
+   wasm_runtime_desc += "\"eos-vm\" : A WebAssembly interpreter providing deterministic and efficient time bound execution with debugging support.\n";
+   delim = ", ";
+#endif
+
+#ifdef EOSIO_EOS_VM_OC_DEVELOPER
+   wasm_runtime_opt += delim + "\"eos-vm-oc\"";
+   wasm_runtime_desc += "\"eos-vm-oc\" : A specialized compiler framework (LLVM) that leverages a multi-pass compilation architecture.\nThe resulting native code from the Optimized Compiler is often an order of magnitude faster thatn others.\n";
```

Comment:
Lets change this to:
"A specialized compiler framework (LLVM) that leverages a multi-pass compilation architecture creating native code.\n"

Reply:
I'd be curious what @arhag had in mind when creating the ticket. Personally I don't think we should mention LLVM here that's too much detail IMO. And I'm not even sure it needs much of a help text because the only way you see this is if you're running an unsupported configuration meant only for us.

Reply:
Changed as @heifner for now.

Reply:
I wasn't even aware of the `eos-vm-oc` option under `--wasm-runtime`. Since that option is not meant to be exposed to the public, I like the approach of only showing the descriptions for `eos-vm` and `eos-vm-jit`.

Reply:
eos-vm-oc is not shown in help text unless it is defined:  #ifdef EOSIO_EOS_VM_OC_DEVELOPER

Reply:
@kimjh2005: That's fine too.

I don't really mind the existing descriptions much. But I will throw in my suggestion into the mix for consideration:

```
"eos-vm" : A WebAssembly interpreter.
"eos-vm-jit" : A WebAssembly runtime that quickly compiles WebAssembly code to native x86 code prior to execution.
"eos-vm-oc" : A WebAssembly runtime that compiles WebAssembly to optimized native x86 code (execution speed of generated native code is faster than code generated by eos-vm-jit but it may block for a significant time during compilation).
```

Reply:
I would suggest modifying @arhag above:
  - Remove `quickly` from `eos-vm-jit`.
  - Remove `execution speed of generated native code is faster than code generated by eos-vm-jit but it ` from `eos-vm-oc`

Reply:
@heifner:

Sure that's fine.

Rewritten with above changes applied for convenience:
```
"eos-vm" : A WebAssembly interpreter.
"eos-vm-jit" : A WebAssembly runtime that compiles WebAssembly code to native x86 code prior to execution.
"eos-vm-oc" : A WebAssembly runtime that compiles WebAssembly to optimized native x86 code (may block for a significant time during compilation).
```

Reply:
> but it may block for a significant time during compilation

That's not the only bad thing that can happen from using eos-vm-oc here; there are consensus ramifications too. Only eosio devs should ever ever ever use that option. IMO don't even document it. Or maybe document it, TOTALLY UNSUPPORTED YOU DON'T WANT THIS, YOU WANT `--enable-eos-vm-oc`

Reply:
@spoonincode Fair enough.

Here are the descriptions I now suggest:
```
"eos-vm" : A WebAssembly interpreter.
"eos-vm-jit" : A WebAssembly runtime that compiles WebAssembly code to native x86 code prior to execution.
"eos-vm-oc" : Unsupported. Use one of the other runtimes along with --enable-eos-vm-oc instead.
```

I do prefer seeing the `eos-vm-oc` option in the list still when `EOSVMOC_ENABLE_DEVELOPER_OPTIONS` is enabled just so we have an easy way of knowing if that flag was used to compile the nodeos binary.

Reply:
--wasm-runtime runtime (=eos-vm-jit)  Override default WASM runtime ( 
                                        "eos-vm-jit", "eos-vm")
                                        "eos-vm-jit" : A WebAssembly runtime 
                                        that compiles WebAssembly code to 
                                        native x86 code prior to execution.
                                        "eos-vm" : A WebAssembly interpreter.

Reply:
Update the PR description.

Reply:
PR is updated.